cluster-autoscaler/cloudprovider/digitalocean/digitalocean_node_group.go

```diff
@@ -292,3 +309,64 @@ func toInstanceStatus(nodeState *godo.KubernetesNodeStatus) *cloudprovider.Insta
 
 	return st
 }
+
+func toNodeInfoTemplate(resp *godo.KubernetesNodePoolTemplateResponse) (*framework.NodeInfo, error) {
+	allocatable, err := parseToQuanitity(resp.Template.Allocatable.CPU, resp.Template.Allocatable.Pods, resp.Template.Allocatable.Memory)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create allocatable resources - %s", err)
+	}
+	capacity, err := parseToQuanitity(resp.Template.Capacity.CPU, resp.Template.Capacity.Pods, resp.Template.Capacity.Memory)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create capacity resources - %s", err)
+	}
+	addedTaints, _, err := taints.ParseTaints(resp.Template.Taints)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse taints from template - %s", err)
+	}
+	l := map[string]string{
+		apiv1.LabelOSStable:   cloudprovider.DefaultOS,
+		apiv1.LabelArchStable: cloudprovider.DefaultArch,
+	}
+
+	l = cloudprovider.JoinStringMaps(l, resp.Template.Labels)
+	node := &apiv1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   generateWorkerName(resp.Name, 1),
+			Labels: l,
+		},
+		Spec: apiv1.NodeSpec{
+			Taints: addedTaints,
+		},
+		Status: apiv1.NodeStatus{
+			Allocatable: allocatable,
+			Capacity:    capacity,
+			Phase:       apiv1.NodeRunning,
+			Conditions:  cloudprovider.BuildReadyConditions(),
+		},
+	}
+	return framework.NewNodeInfo(node, nil), nil
+}
+
+func parseToQuanitity(cpu int64, pods int64, memory string) (apiv1.ResourceList, error) {
+	c := resource.NewQuantity(cpu, resource.DecimalSI)
+	p := resource.NewQuantity(pods, resource.DecimalSI)
+	m, err := resource.ParseQuantity(memory)
+	if err != nil {
+		return nil, err
+	}
+	return apiv1.ResourceList{
+		apiv1.ResourceCPU:    *c,
+		apiv1.ResourceMemory: m,
+		apiv1.ResourcePods:   *p,
+	}, nil
+}
+
+func generateWorkerName(poolName string, workerID int64) string {
+	var customAlphabet string = "n38uc7mqfyxojrbwgea6tl2ps5kh4ivd01z9"
+	var customAlphabetSize int64 = int64(len(customAlphabet))
+	s := ""
+	for ; workerID > 0; workerID = workerID / customAlphabetSize {
+		s = string(customAlphabet[workerID%customAlphabetSize]) + s
+	}
```

Comment:
I can't quite follow the logic of this. The worker ID seems to be intended to be something dynamic.
However the only call I can find is for the worker ID to be `1`.
If you just want to create a random fixed size string suffix from a defined charset, you could follow something like this https://www.calhoun.io/creating-random-strings-in-go/